include/miral/miral/application_switcher.h

```diff
@@ -89,6 +97,67 @@ class ApplicationSwitcher
     class Self;
     std::shared_ptr<Self> self;
 };
+
+/// This class provides a common shortcut strategy for the
+/// #miral::ApplicationSwitcher.
+///
+/// By default, this class will do the following:
+/// - Call #miral::ApplicationSwitcher::next_app on `alt + tab`.
+/// - Call #miral::ApplicationSwitcher::prev_app on `alt + shift + tab`
+/// - Call #miral::ApplicationSwitcher::next_window on `alt + grave`
+/// - Call #miral::ApplicationSwitcher::prev_window` on `alt + shift + grave`.
+///
+/// This class should be provided to a #miral::MirRunner::run_with.
+///
+/// \remark Since MirAL 5.6
+class ApplicationSwitcherKeyboardFilter
+{
+public:
+    /// Describes the keyboard shortcuts for the application switcher keyboard filter.
+    struct KeybindConfiguration
+    {
+        /// This is the modifier that must be held for any application switcher
+        /// action to begin.
+        ///
+        /// Defaults to #mir_input_event_modifier_alt.
+        MirInputEventModifier primary_modifier = mir_input_event_modifier_alt;
+
+        /// This is the modifier that must be held alongside the primary modifier
+        /// in order to reverse the direction of an action being run. For example,
+        /// holding the primary modifier and hitting the application key will select
+        /// the next application, while holding both the primary and reverse modifiers
+        /// and hitting the application will select the previous application.
+        ///
+        /// Defaults to #mir_input_event_modifier_shift.
+        MirInputEventModifier reverse_modifier = mir_input_event_modifier_shift;
+
+        /// This is the key that must be clicked in order to trigger either
+        /// #miral::ApplicationSwitcher::next_app or #miral::ApplicationSwitcher::prev_app,
+        /// depending on the modifiers that are currently being held.
+        ///
+        /// Defaults to KEY_TAB.
+        int application_key = KEY_TAB;
+
+        /// This is the key that must be clicked in order to trigger either
+        /// #miral::ApplicationSwitcher::next_window or #miral::ApplicationSwitcher::prev_window,
+        /// depending on the modifiers that are currently being held.
+        ///
+        /// Defaults to KEY_GRAVE.
+        int window_key = KEY_GRAVE;
+    };
+
+    /// Create a new keyboard filter for the provided \p switcher with the provided
+    /// \p keybind_configuration.
+    ApplicationSwitcherKeyboardFilter(ApplicationSwitcher const& switcher, KeybindConfiguration const& keybind_configuration);
```

Comment:
Is there any value in forcing the user to create and pass the `switcher`?

Can't we encapsulate the remaining "boilerplate" the user needs? (Starting the internal client, registering the stop callback, etc)